File: Week 4/Monday/FlagsC/FlagsC/Model/StateCollectionViewCell.m
//
//  StateCollectionViewCell.m
//  FlagsC
//
//  Created by Thomas Cowern New on 12/16/18.
//  Copyright © 2018 Thomas Cowern New. All rights reserved.
//

#import "StateCollectionViewCell.h"
#import "State.h"

@implementation StateCollectionViewCell

- (void) setState:(State *)state
{
    _state = state;
    [self updateWithState: state];
}

- (void) updateWithState:(State *)state
{
    if (state) {
        self.flagImageview.image = [UIImage imageNamed: state.abbreviation];
        self.flagLabel.text = state.name;
    }
}

@end

File: Week 4/Monday/FlagsSwift/FlagsSwift/Resources/StateCollectionViewCell.swift
//
//  StateCollectionViewCell.swift
//  FlagsSwift
//
//  Created by Thomas Cowern New on 12/16/18.
//  Copyright © 2018 Thomas Cowern New. All rights reserved.
//

import UIKit

class StateCollectionViewCell: UICollectionViewCell {
    
    @IBOutlet weak var flagView: UIImageView!
    @IBOutlet weak var flagLabel: UILabel!
    
    var state: State? {
        didSet {
            updateViews()
        }
    }
    
    func updateViews() {
        guard let state = state else { return }
        flagView.image = UIImage(named: state.abbreviation)
        flagLabel.text = state.name
    }
}
